pkg/api/pod/util.go

```diff
@@ -813,3 +818,50 @@ func setsWindowsHostProcess(podSpec *api.PodSpec) bool {
 
 	return inUse
 }
+
+func ValidatePodTemplateSpecAllowInValidLabelValueInSelector(spec *api.PodSpec) bool {
+	if spec.Affinity != nil {
+		return ValidateAffinityAllowInValidLabelValueInSelector(spec.Affinity)
+	}
```

Comment:
if we're not using ValidateAffinityAllowInValidLabelValueInSelector anywhere else, would it be simpler to inline this?

```
func hasInvalidLabelValueInAffinitySelector(spec *api.PodSpec) bool {
  if spec == nil || spec.Affinity == nil {
    return false
  }
  if spec.Affinity.PodAffinity != nil {
    for _, term := range affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
      ...
    }
    for _, term := range affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution {
      ...
    }
  }
  if spec.Affinity.PodAntiAffinity != nil {
    ...
  }
  return false
}
```

Reply:
SGTM, I will update the PR